examples/Nat/TcpState.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using PacketDotNet;
+
+namespace Pax.Examples.Nat
+{
+  internal class TcpState : ITransportState<TcpPacket>
+  {
+    private TcpDirectionalState InOutConnection = TcpDirectionalState.None;
+    private TcpDirectionalState OutInConnection = TcpDirectionalState.None;
+    public bool ClosedFromInside { get { return InOutConnection == TcpDirectionalState.None || InOutConnection == TcpDirectionalState.FinAck; } }
+    public bool ClosedFromOutside { get { return OutInConnection == TcpDirectionalState.None || OutInConnection == TcpDirectionalState.FinAck; } }
+    public bool CanBeClosed { get { return ClosedFromInside && ClosedFromOutside; } }
+
+    public void UpdateState(TcpPacket packet, bool packetFromInside)
```

Comment:
What happens if we receive a repeated `Syn` travelling in the same direction? (From inside to out, say) Is the machine reset? That is, can evil person (call them E) break user A's connection to outside server B, when E sends the NAT a Syn addressed to B (and spoofed to appear to come from A)?